src/harness/BenchmarkDotNet.Extensions/CommandLineOptions.cs

```diff
@@ -53,6 +54,44 @@ public static List<string> ParseAndRemoveStringsParameter(List<string> argsList,
             return argsList;
         }
 
+        public static List<string> ParseAndRemovePairsParameter(List<string> argsList, string parameter, out Dictionary<string, object> pairValues)
+        {
+            int parameterIndex = argsList.IndexOf(parameter);
+            pairValues = new Dictionary<string, object>();
+
+            if (parameterIndex + 1 < argsList.Count)
+            {
+                while (parameterIndex + 1 < argsList.Count && !argsList[parameterIndex + 1].StartsWith("-"))
+                {
+                    string[] pair = argsList[parameterIndex + 1].Split(':');
+                    object value;
+                    int i_result;
+                    bool b_result;
+                    if (Int32.TryParse(pair[1], out i_result))
```

Comment:
If you use this approach to compare the types of the parameters, you will need to make sure that every type that we can use here is covered. Otherwise, if we ever have a type like byte, or long this piece of logic would fail.